src/SFA.DAS.EAS.Portal/Application/Services/AccountDocumentService.cs

```diff
@@ -20,6 +20,17 @@ public Task<AccountDocument> Get(long id, CancellationToken cancellationToken =
                .SingleOrDefaultAsync(a => a.AccountId == id, cancellationToken);            
         }
 
+        public async Task<AccountDocument> GetOrCreate(long id, CancellationToken cancellationToken = default)
```

Comment:
Would it be worth splitting the Get/GetOrCreate and the Save out into separate interfaces to stop duplication of code?